Signal/src/MultiDeviceProfileKeyUpdateJob.swift

```diff
@@ -0,0 +1,58 @@
+//
+//  Copyright (c) 2017 Open Whisper Systems. All rights reserved.
+//
+
+import Foundation
+import PromiseKit
+
+/**
```

Comment:
Following Signal-Android's lead - instead of attaching our profile key to *every* sync message, we send it explicitly when updating our profile.

This is less fool-proof, but it did seem like overkill to transmit an extra 30-some bytes on every sync message, when we ultimately only need to transmit it once. Especially given how chatty sync messages are.

Reply:
Hmm, but we could link to desktop at any time and years could pass after linking before users update their profile again.  What if we also include it in the "bulk contact sync" messages?

Reply:
Profile key is sent as part of provisioning. This job just exists for legacy clients.

Reply:
Gotcha thanks.